core/usig-ui.go

```diff
@@ -18,20 +18,29 @@ package minbft
 
 import (
 	"fmt"
-	"sync"
 
 	"github.com/nec-blockchain/minbft/api"
+	"github.com/nec-blockchain/minbft/core/internal/peerstate"
 	"github.com/nec-blockchain/minbft/messages"
 	"github.com/nec-blockchain/minbft/usig"
 )
 
-// uiAcceptor verifies and accepts a unique identifier of a message.
+// uiCapturer verifies USIG unique identifier of message, and
+// synchronizes beginning of sequential processing of it.
 //
-// A UI is accepted if it is valid for the message and all previous
-// UIs generated by the replica USIG instance have already been
-// accepted. The UI is new if it has been accepted for the first time.
-// This function is safe to invoke concurrently.
-type uiAcceptor func(msg messages.MessageWithUI) (new bool, err error)
+// The sequential processing is achieved by waiting until uiReleaser
+// is invoked on a message with a UI generated by the same replica,
+// that has the preceding counter value. The return value new
+// indicates if the UI has not been processed before. It is safe to
+// invoke concurrently.
```

Comment:
Mention the guarantee that the same UI from the same replica cannot be captured concurrently until it gets released.